torch_geometric/loader/node_loader.py

```diff
@@ -0,0 +1,154 @@
+from typing import Any, Callable, Dict, Iterator, Optional, Tuple, Union
+
+import torch
+
+from torch_geometric.data import Data, HeteroData
+from torch_geometric.data.feature_store import FeatureStore
+from torch_geometric.data.graph_store import GraphStore
+from torch_geometric.loader.base import DataLoaderIterator
+from torch_geometric.loader.utils import (
+    filter_custom_store,
+    filter_data,
+    filter_hetero_data,
+    get_input_nodes,
+)
+from torch_geometric.sampler.base import (
+    BaseSampler,
+    HeteroSamplerOutput,
+    NodeSamplerInput,
+    SamplerOutput,
+)
+from torch_geometric.typing import InputNodes
+
+
+class NodeLoader(torch.utils.data.DataLoader):
+    r"""A data loader that performs neighbor sampling from node information,
+    using a generic :class:`~torch_geometric.sampler.BaseSampler`
+    implementation that defines a `sample_from_nodes` function and is supported
+    on the provided input :obj:`data` object.
+
+    Args:
+        data (torch_geometric.data.Data or torch_geometric.data.HeteroData):
+            The :class:`~torch_geometric.data.Data` or
+            :class:`~torch_geometric.data.HeteroData` graph object.
+        node_sampler (torch_geometric.sampler.BaseSampler, optional): The
+            sampler implementation to be used with this loader. Note that the
+            sampler implementation must be compatible with the input data
+            object.
+        node_sampler_kwargs (Dict[str, Any], optional): the keyword arguments
+            to be passed to the constructor of `node_sampler`; only necessary
+            if `node_sampler` has not already been constructed.
+        initialize_sampler (bool, optional): whether `node_sampler` should be
```

Comment:
I don't necessarily understand this argument. Why not just always require a `node_sampler` argument and drop `node_sampler_kwargs` and `initialize_sampler`. I feel like this makes the base class interface more complex than it needs to be.

Reply:
Agreed, fixed - thanks!